app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -379,7 +446,11 @@ public void updateAccount(Account account, boolean allowIdentifierUpdates) {
         accountToUpdate.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
 
         // Update
-        hibernateHelper.update(accountToUpdate);
+        try {
+            hibernateHelper.update(accountToUpdate);            
+        } catch(PersistenceException pe) {
+            throw convertPersistenceException(pe, persistedAccount.getId());    
+        }
```

Comment:
If you update an account to add an email/phone/extID, the error message will point to this account and not to the conflicting account. Is this the expected behavior?

Reply:
Hmm... no. I will change this.